bytecomp/translmod.ml

```diff
@@ -1323,12 +1341,18 @@ let transl_store_package component_names target_name coercion =
 
 open Format
 
+let print_cycle ppf =
+  Format.pp_print_list ~pp_sep:(fun ppf () -> fprintf ppf "@ -> ")
+    Printtyp.ident ppf
+
 let report_error ppf = function
-    Circular_dependency id ->
+    Circular_dependency cycle ->
+      let[@manual.ref "s-recursive-modules"] chapter, section = 8, 4 in
       fprintf ppf
-        "@[Cannot safely evaluate the definition@ \
-         of the recursively-defined module %a@]"
-        Printtyp.ident id
+        "@[Cannot safely evaluate the definition of the following cycle@ \
+         of recursively-defined modules:@ %a.@ \
+         There is no safe modules in this cycle@ (see manual section %d.%d)@]"
```

Comment:
I think this should be "There are no safe modules" or "There is no safe module".

Reply:
You are right. Fixed, thanks!